lib/storage/bucket.js

```diff
@@ -252,16 +274,18 @@ Bucket.prototype.getFiles = function(query, callback) {
       callback(err);
       return;
     }
+
     var files = (resp.items || []).map(function(item) {
       var file = that.file(item.name);
       file.metadata = item;
       return file;
     });
+    var prefixes = resp.prefixes || null;
     var nextQuery = null;
     if (resp.nextPageToken) {
       nextQuery = extend({}, query, { pageToken: resp.nextPageToken });
     }
-    callback(null, files, nextQuery);
+    callback(null, files, nextQuery, prefixes);
```

Comment:
My thoughts are we should provide the prefixes before `nextQuery` i.e. `nextQuery` should always be the last object however this will result in a breaking change so I understand why it was put here for now. Thinking ahead, are there any other possibilities of different objects that can be returned such that it may be better to return in an object rather than polluting the callback parameters will a large chain of different values? Looking at [the docs](https://cloud.google.com/storage/docs/json_api/v1/objects/list) it appears not but typically I try to stick to a maximum of 3 parameters per function (callback or otherwise)... after that, it's hard to remember the order of parameters.

Pinging @stephenplusplus for additional thoughts here.